### DISCUSSION TITLE
Nats micro with Jetstream

### DISCUSSION THREAD

**Discussion by suikast42** [nats-io/nats-server#4984] [General]
Nats micro looks very well for managing and deploying services  over nats. 

But as I understand the hanlders are registrered as a "persistence less" listener to a subject. 

It could be very handy to combine nats micro with jetstream. The handler should be able to ack and nack messages for handling redelivery.

**Comment by ripienaar** [comment]
Something like this is roughly planned for a future itteration, agree it would be useful.  No immediate plans though

**Reply by dnbsd** [reply]
Hello @ripienaar, I was wondering if there’s been any update since your comment?

**Reply by nickchomey** [reply]
I'd also be very interested in this. I just built a mechanism on micro only to realize it didn't have persistence. So had to build a separate one that used jetstream and a replyto subject header, with a separate subscriber for the response on that subject.

**Reply by ripienaar** [reply]
Not yet no. It’s fairly easy to do for a particular case but solving it generically for everyone is quite hard. 

Though the missing piece was per message TTLs which we do now have so it’s more likely we will be able to do something good here. 

Still not on the immediate roadmap
